docs/source/contributing/layout.rst

```diff
@@ -0,0 +1,126 @@
+===============================
+Layout of JupyterHub Repository
+===============================
+
+This page covers the layout and structure of the JupyterHub repository. It provides brief information on the purpose of the various files and directories in the repository. Please consult JupyterHub documentation for more details.
+
+
+Directory Structure
+===================
+
+ci/
+    This directory contains files for automating aspects of the continuous integration pipeline of the JupterHub project; some of which includes checking distribution packages to ensure they contain all expected files.
+
+demo-image/
+    This directory contains a Dockerfile for building a JupyterHub image meant for demo or testing purposes only. You can check out the `README` file in the directory for more details.
+
+dockerfiles/
+    This directory contains the Dockerfile for building the JupyterHub base image. You can check out the `README` in the directory for more details.
+
+docs/
+    This directory contains the documentation files for the JupyterHub project.
+
+examples/
+    This directory details several services provided by the JupyterHub project, alongside their configurations to enable you configure JupyterHub to use those services.
+
+jsx/
+    This directory contains current updates to the JupyterHub Admin Dashboard.
+
+JupyterHub/
+    This directory contains the core code files for the JupyterHub project.
+
+onbuild/
+    This directory contains the Dockerfile for building a JupyterHub image. Specifically, it would include the configuration file(`JupyterHub_config.py`) in the image build.
+
+share/jupyterHub/
+    This directory contains JupyterHub data and static files.
+
+singleuser/
+    This directory contains the Dockerfile for building a JupyterHub image for a single user. You can check out the `README` in the directory for more details.
+
+testing/
+    This directory contains a sample JupyterHub configuration file for testing.
+
+Executable files
```

Comment:
These aren't executables, but I'm not sure what exactly the best name is. Configuration files? Packaging files?

Reply:
@minrk  the naming of the heading is actually based off a similar documentation written  by @willingc 
https://github.com/openhatch/oh-mainline/blob/master/LAYOUT#L4

Reply:
And yeah, I could still do that if you want. A layout of the code in `jupyterhub/`. I would just have it added as  subs under the `jupyterhub` dir currently in the layout. What do you think?

Reply:
Hi @minrk I just made a new commit as regards this https://github.com/jupyterhub/jupyterhub/pull/4066/commits/4eb13d5eafac1af925ce310ff7d244d239c42dbf